gravityflow/gravityflow
# Webhook Step - Add support for GET to select fields to query string

## Description
Refer to [HS #12026](https://secure.helpscout.net/conversation/1029609864/12026?folderId=1776095). If wp_remote_request is passed args for a GET request with a body (array), it will convert the key/values into querystring parameters. If the URL field already includes querystring values, the wp_remote_request logic will identify and append the chosen body fields.

This PR exposes the Request Field Values step setting for GET request types. Previously, the user would need to use multiple merge tags that is a bad UI for more than 1 or 2 fields of data. The choice to not display All fields or Request Type (Raw/Select Fields) is intentional as the querystring parameters are generally used in a one-dimensional context. For more advanced cases, user can use code-based approach with gravityflow_webhook_args filter.


## Testing instructions
- Create new outgoing webhook steps and customize existing ones.
- Confirm that UI to define a GET with body converted to querystring works as expected.
- Confirm that UI not setting a body/querystring is not affected.
- Confirm that other request methods (POST, PUT, etc) for webhooks continue to be submit.

## Automated Test Enhancements
**In Progress** Add to test 0027 for field mapping to querystring values
 
## Screenshots
![image](https://user-images.githubusercontent.com/4549984/71685176-82f0fe00-2d65-11ea-8a8a-7d8ab8b76bca.png)


## Documentation Changes?
- Update Outgoing Webook support pages.

## Checklist:
